java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -602,6 +606,175 @@ public static CompletableFuture<Object> arrinsert(
+     *     <pre>{@code
+     * Json.set(client, key, "$", "{\"a\": [\"value\", 3], \"b\": {\"a\": [3, [\"value\", false], 5]}}").get();
+     * Object[] result1 = Json.arrindex(client, key, "$..a", new JsonScalar(3)).get();
+     * assertArrayEquals(new Object[] {Long.valueOf(1), Long.valueOf(0)}, result1);
+     *
+     * Object result2 = Json.arrindex(client, key, $.a, new JsonScalar(3)).get();
+     * assertEquals(Long.valueOf(1), result2);
+     * }</pre>
+     */
+    public static CompletableFuture<Object[]> arrindex(
+            @NonNull BaseClient client,
+            @NonNull String key,
+            @NonNull String path,
+            @NonNull JsonScalar scalar,
+            @NonNull JsonIndexOptions options) {
+
+        return executeCommand(
+                client,
+                new ArgsBuilder()
+                        .add(JSON_ARRINDEX)
+                        .add(key)
+                        .add(path)
+                        .add(scalar.toString())
+                        .add(options.toArgs())
+                        .toArray());
+    }
+
+    /**
+     * Searches for the first occurrence of a <code>scalar</code> JSON value in the arrays at the
+     * path.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @param path The path within the JSON document.
+     * @param scalar The scalar value to search for; JSON scalar refers to values that are not objects
+     *     or arrays.
+     * @return
+     *     <ul>
+     *       <li>For JSONPath (<code>path</code> starts with <code>$</code>): Returns an array with a
+     *           list of integers for every possible path, indicating the index of the matching
+     *           element. The value is <code>-1</code> if not found. If a value is not an array, its
+     *           corresponding return value is <code>null</code>..
+     *       <li>For legacy path (path doesn't start with <code>$</code>): Returns an integer
+     *           representing the index of matching element, or <code>-1</code> if not found. If the
+     *           value at the <code>path</code> is not an array, an error is raised.
+     *     </ul>
+     *
+     * @example
+     *     <pre>{@code
+     * Json.set(client, key, "$", "{\"a\": [\"value\", 3], \"b\": {\"a\": [3, [\"value\", false], 5]}}").get();
+     * Object[] result1 = Json.arrindex(client, gs(key), gs("$..a"), new JsonScalar(3)).get();
+     * assertArrayEquals(new Object[] {Long.valueOf(1), Long.valueOf(0)}, result1);
+     *
+     * Object result2 = Json.arrindex(client, gs(key), gs($.a), new JsonScalar(3)).get();
+     * assertEquals(Long.valueOf(1), result2);
+     * }</pre>
+     */
+    public static CompletableFuture<Object[]> arrindex(
+            @NonNull BaseClient client,
+            @NonNull GlideString key,
+            @NonNull GlideString path,
+            @NonNull JsonScalar scalar,
+            @NonNull JsonIndexOptionsBinary options) {
```

Comment:
Why do we have 2 separate classes for Binary and String?

Reply:
toArgs - one for glidestring and one for normal string

Reply:
IT's the Java way™